posthog/helpers/session_recording.py

```diff
@@ -24,15 +24,16 @@ class PaginatedSnapshotList:
 
 def preprocess_session_recording_events(events: List[Event]) -> List[Event]:
     result = []
-    snapshots_by_session = defaultdict(list)
+    snapshots_by_session_and_window_id = defaultdict(list)
     for event in events:
         if is_unchunked_snapshot(event):
-            session_recording_id = event["properties"]["$session_id"]
-            snapshots_by_session[session_recording_id].append(event)
+            session_id = event["properties"]["$session_id"]
+            window_id = event["properties"].get("$window_id")
+            snapshots_by_session_and_window_id[f"{session_id}-{window_id}"].append(event)
```

Comment:
Python dicts can use tuples as keys. Use that instead.

Reply:
Oooo had no idea - thanks for the tip!